homeassistant/components/device_tracker/owntracks.py

```diff
@@ -85,7 +89,12 @@ def owntracks_event_update(topic, payload, qos):
         if data['event'] == 'enter':
             zone = hass.states.get("zone.{}".format(location))
             with LOCK:
-                if zone is not None:
+                if zone is None:
```

Comment:
I think this is right. If you enter a beacon region that matches a HA zone - then you should change the user's location to that beacon's location. So In my case when I enter my drive and detect the beacon - my location should be set to zone.drive. This is unchanged from before.

If HA doesn't have a zone that matches the beacon - then that means it's a mobile beacon - and you add it to the list of beacon's to set to the user's location.

Worst case is you have a mobile beacon in your car - and enter a beacon with a zone in HA. In that case both the mobile beacon and user are set to the zone beacon's location.

I will write something about how to set this up.

Does this make sense to you?